src/holisticai/utils/plots/_plots.py

```diff
@@ -0,0 +1,223 @@
+"""
+This module implements functions for visualizing 2D datasets and their properties in
+the context of dataset shift. These functions highlight different aspects of the data,
+such as test set points, predicted values, and neighborhoods around specific points of
+interest.
+
+Functions included:
+-------------------
+- plot_graph: Plots a 2D scatter plot of the entire dataset.
+- plot_highlight_test_set: Highlights a specific subset of points in the dataset.
+- plot_just_test_set: Annotates all points with their indices in the dataset.
+- plot_ytest_ypred: Displays both true and predicted labels on the same plot with a
+slight vertical offset.
+- plot_neighborhoods: Plots the neighborhoods around specified points of interest and
+calculates accuracy.
+"""
+
+
+def plot_graph(X, y):
+    """
+    Plots a 2D scatter plot of a dataset.
+
+    Parameters:
+    ----------
+    X : np.ndarray
+        The feature matrix where each row represents a sample and each column represents a feature.
+    y : np.ndarray
+        The labels for each sample.
+
+    Returns:
+    -------
+    None
+    """
+    import matplotlib.pyplot as plt
+
+    # Plotting the data
+    plt.figure(figsize=(8, 6))
+    plt.scatter(X[:, 0], X[:, 1], c=y, cmap="viridis", s=50, edgecolor="k")
+
+    plt.xlabel("Feature 1")
+    plt.ylabel("Feature 2")
+    plt.title("Your 2D Dataset")
+    plt.show()
+
+
+def plot_highlight_test_set(X, y, test_indices):
```

Comment:
i think we can merge this function with plot_graph and use test_indices as optional parameter

Reply:
Done.